packages/blade/src/components/AutoComplete/_decisions/decisions.md

```diff
@@ -0,0 +1,240 @@
+| onSelectChange     | Callback when selected items change                                                                                  | ({ name, values }: { name: string; values: string[] } ) => void |                   |
+
+## Usage
+
+AutoComplete will be built as a new trigger for existing Dropdown component.
+
+Depending on where and how AutoComplete is used, the API can differ. Checkout examples below on how code will look like for a particular usage.
+
+- [Basic Usage](#basic-usage)
+- [Controlled AutoComplete](#controlled-autocomplete)
+- [In BottomSheet](#in-bottomsheet)
+- [With Tags Outside](#with-tags-outside)
+- [With Different Inactive - Active States](#with-different-inactive---active-statess)
+
+### Basic Usage
+
+```jsx
+<Dropdown selectionType="multiple">
+  <AutoComplete label="Select Cities" />
+  <DropdownOverlay>
+    <ActionList>
+      <ActionListItem title="Mumbai" value="mumbai" />
+      <ActionListItem title="Pune" value="pune" />
+      <ActionListItem title="Hyderabad" value="hyderabad" />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>
+```
+
+### Controlled AutoComplete
+
+There are 3 controlled states that are relevant to AutoComplete
+
+1. Controlled state of filtering items
+2. Controlled state of the Input where user types
+3. Controlled state of selected items
+
+```jsx
+<AutoComplete
+  // For controlling the input value
+  value={}
+  onChange={(inputValue) => {}}
+  // For controlling the selections
+  selected={}
+  onSelectChange={({ name, values }) => {}}
+/>
+
+// ...
+
+// Filtering can be controlled by looping over <ActionListItem />
+<ActionList>
+  {cities.map((city) => <ActionListItem title={city} />)}
+</ActionList>
+```
+
+Complex flows such as [Adding New Item to ActionList](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=47091%3A711978&mode=dev) should be handled with Controlled API.
+
+<details>
+<summary>Show Full Code</summary>
+
+```jsx
+const allCities = ['Mumbai', 'Pune', 'Hyderabad'];
+
+const ControlledAutoComplete = () => {
+  // Controlled Filtering
+  const [cities, setCities] = React.useState(allCities);
+  // Controlled Input Value
+  const [currentInputValue, setCurrentInputValue] = React.useState('');
+  // Controlled Selections
+  const [currentSelections, setCurrentSelections] = React.useState<string[]>([]);
+
+  return (
+    <Dropdown>
+      <AutoComplete
+        label="Select City"
+        // Controlled input value
+        // Sets the input value of input element in AutoComplete
+        value={currentInputValue}
+        // Triggers when input value changes
+        onChange={(inputValue) => {
+          if (inputValue) {
+            // filtering logic
+            setCities(allCities.filter((city) => city.includes(inputValue)));
```

Comment:
We can probably have an example with derived state, instead of manually setting filtered items. 

Similar to how ariakit is doing: https://ariakit.org/examples/combobox-filtering#filtering-items

> Yes just same thing but two ways of doing but our examples are probably people will copy paste.

Reply:
Can you explain in more React-noob terms? I didn't understand that ariakit example 🙈